python/sidecar/http.py

```diff
@@ -28,62 +29,192 @@
+
+
+@project_api.post("/")
+async def create_project(project_path: str = None):
+    """
+    Creates a project directory in the filesystem
+
+    Parameters
+    ----------
+    project_path : str
+        The path to the project directory. Only necessary for Desktop.
+        For web, the project is stored in a private directory on the server.
+    """
+    user_id = "user1"
+    project_id = str(uuid4())
+    project_path = projects.create_project(user_id, project_id, project_path)
+    return {
+        project_id: project_path,
+    }
+
+
+@project_api.get("/{project_id}")
+async def get_project(project_id: str):
+    """
+    Returns the project path and a list of files in the project
+    """
+    user_id = "user1"
+    project_path = projects.get_project_path(user_id, project_id)
+    filepaths = projects.get_project_files(user_id, project_id)
+    return {
+        "project_id": project_id,
+        "project_path": project_path,
+        "filepaths": filepaths,
+    }
+
+
+@project_api.delete("/{project_id}")
+async def delete_project(project_id: str):
+    """
+    Deletes a project directory and all files in it
+    """
+    user_id = "user1"
+    projects.delete_project(user_id, project_id)
+    return {
+        "status": "success",
+        "message": "Project deleted",
+        "project_id": project_id,
+    }
+
+
+@project_api.get("/{project_id}/files")
+async def get_project_files(project_id: str):
+    user_id = "user1"
+    filepaths = projects.get_project_files(user_id, project_id)
+    return filepaths
+
+
+
+# Filesystem API
+# --------------
+@filesystem_api.put("/{project_id}/{filepath:path}")
+async def create_file(project_id: str, filepath: Path, file: UploadFile = File(...)):
+    user_id = "user1"
+    project_path = projects.get_project_path(user_id, project_id)
+    filepath = project_path / filepath
+
+    if not filepath.exists():
+        filepath.parent.mkdir(parents=True, exist_ok=True)
+    
+    try:
+        with open(filepath, "wb") as f:
+            shutil.copyfileobj(file.file, f)
+    except Exception as e:
+        print(e)
+    finally:
+        file.file.close()
+    return {
+        "status": "success",
+        "message": "File uploaded",
+        "filepath": filepath,
+    }
+
+
+@filesystem_api.get("/{project_id}/{filepath:path}")
+async def read_file(project_id: str, filepath: Path):
+    user_id = "user1"
+    project_path = projects.get_project_path(user_id, project_id)
+    filepath = project_path / filepath
+
+    if not filepath.exists():
+        return {
+            "status": "error",
+            "message": "File not found",
+            "filepath": filepath,
+        }
+    return FileResponse(filepath)
+
+
+@filesystem_api.delete("/{project_id}/{filepath:path}")
+async def delete_file(project_id: str, filepath: Path):
+    user_id = "user1"
+    project_path = projects.get_project_path(user_id, project_id)
+    filepath = project_path / filepath
+    try:
+        os.remove(filepath)
```

Comment:
@cguedes I ended up deleting `filesystem.py` because once I cleaned this PR up, this line was literally the only one remaining that was actually used. FastAPI does a nice job of having built-in functionality for reading files with `FileResponse` and uploading files with `UploadFile`.